app/models/notification.rb

```diff
@@ -252,6 +252,29 @@ def send_tag_adjustment_notification(tag_adjustment)
     end
     handle_asynchronously :send_tag_adjustment_notification
 
+    def send_milestone_notification(milestone_hash)
```

Comment:
I would suggest moving the send notification code to a service object not to overload the model.

Reply:
Yeah, I agree. I think I'll move all of that code in a different PR.